### ISSUE TITLE
Manhattan plotting issue

### ISSUE THREAD

**Issue by swvanderlaan** [Cloufield/gwaslab#42]
This is my command:

```
# manhattan and qq plot
ea_aa_sumstats.plot_mqq(
    skip=2,
    cut=20,
    sig_line=True,
    sig_level=5e-8,  
    anno="GENENAME",
    anno_style="right",
    windowsizekb=500,
    arm_offset=2,
    repel_force=0.01,  # default 0.01
    use_rank=True,
    build="19",
    # mode="m",
    stratified=True,
    figargs={"figsize": (25, 15), "dpi": 100},
    title="Multi-ancestry GWAS on Phenotype",
    save=os.path.join(PLOTS_loc, "manhattan.50kb.300dpi.pdf"),
    saveargs={"dpi": 300},
    verbose=True,
)
```

And this is the error I get. What am I missing here?

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[47], line 2
      1 # manhattan and qq plot
----> 2 ea_aa_sumstats.plot_mqq(
      3     skip=2,
      4     cut=20,
      5     sig_line=True,
      6     sig_level=5e-8,  
      7     anno="GENENAME",
      8     anno_style="right",
      9     windowsizekb=500,
     10     arm_offset=2,
     11     repel_force=0.01,  # default 0.01
     12     use_rank=True,
     13     build="19",
     14     # mode="m",
     15     stratified=True,
     16     figargs={"figsize": (25, 15), "dpi": 100},
     17     title="Multi-ancestry GWAS on Phenotype",
     18     save=os.path.join(PLOTS_loc, "manhattan.50kb.300dpi.pdf"),
     19     saveargs={"dpi": 300},
     20     verbose=True,
     21 )

File [~/mambaforge/envs/gwas/lib/python3.8/site-packages/gwaslab/Sumstats.py:460](https://file+.vscode-resource.vscode-cdn.net/Users/username/git/data/gwas/~/mambaforge/envs/gwas/lib/python3.8/site-packages/gwaslab/Sumstats.py:460), in Sumstats.plot_mqq(self, build, **args)
    457 if build is None:
    458     build = self.meta["gwaslab"]["genome_build"]
--> 460 plot = mqqplot(self.data,
    461                snpid=snpid, 
    462                chrom=chrom, 
    463                pos=pos, 
    464                p=p, 
    465                eaf=eaf,
    466                build = build, 
    467                **args)
    469 return plot

File [~/mambaforge/envs/gwas/lib/python3.8/site-packages/gwaslab/mqqplot.py:559](https://file+.vscode-resource.vscode-cdn.net/Users/username/git/data/gwas/~/mambaforge/envs/gwas/lib/python3.8/site-packages/gwaslab/mqqplot.py:559), in mqqplot(insumstats, chrom, pos, p, snpid, eaf, ea, nea, chr_dict, xtick_chr_dict, vcf_path, vcf_chr_dict, gtf_path, gtf_chr_dict, gtf_gene_name, rr_path, rr_header_dict, rr_chr_dict, rr_lim, mlog10p, scaled, mode, scatter_kwargs, qq_scatter_kwargs, qq_line_color, region, region_ref, region_ref2, region_step, region_grid, region_grid_line, region_lead_grid, region_lead_grid_line, region_hspace, region_ld_threshold, region_ld_colors, region_ld_colors1, region_ld_colors2, region_recombination, region_protein_coding, region_flank_factor, region_anno_bbox_args, taf, tabix, mqqratio, bwindowsizekb, density_color, density_range, density_trange, density_threshold, density_tpalette, density_palette, windowsizekb, anno, anno_set, anno_alias, anno_d, anno_args, anno_style, anno_fixed_arm_length, anno_source, anno_adjust, anno_max_iter, arm_offset, arm_scale, arm_scale_d, cut, skip, ystep, ylabels, ytick3, cutfactor, cut_line_color, cut_log, jagged, jagged_len, jagged_wid, sig_line, sig_level, sig_level_lead, sig_line_color, suggestive_sig_line, suggestive_sig_level, suggestive_sig_line_color, additional_line, additional_line_color, sc_linewidth, highlight, highlight_color, highlight_windowkb, highlight_anno_args, pinpoint, pinpoint_color, stratified, maf_bins, maf_bin_colors, gc, include_chrXYMT, ylim, xpad, chrpad, drop_chr_start, title, mtitle, qtitle, title_pad, title_fontsize, fontsize, font_family, anno_fontsize, figargs, colors, marker_size, use_rank, verbose, repel_force, build, dpi, save, saveargs, _invert, expected_min_mlog10p, log)
    555 #sort & add id
    556 ## Manhatann plot ###################################################
    557 if ("m" in mode) or ("r" in mode): 
    558     # assign index i and tick position
--> 559     sumstats,chrom_df=_quick_assign_i_with_rank(sumstats, chrpad=chrpad, use_rank=use_rank, chrom="CHR",pos="POS",drop_chr_start=drop_chr_start)
    561     ## Assign marker size ##############################################
    562     sumstats["s"]=1

File [~/mambaforge/envs/gwas/lib/python3.8/site-packages/gwaslab/quickfix.py:190](https://file+.vscode-resource.vscode-cdn.net/Users/username/git/data/gwas/~/mambaforge/envs/gwas/lib/python3.8/site-packages/gwaslab/quickfix.py:190), in _quick_assign_i_with_rank(sumstats, chrpad, use_rank, chrom, pos, drop_chr_start)
    187     posdiccul[i]= posdiccul[i-1] + posdiccul[i] + sumstats[pos].max()*chrpad
    189 # convert base pair postion to x axis position using the cumulative sum dictionary
--> 190 sumstats["add"]=sumstats[chrom].apply(lambda x : posdiccul[int(x)-1])
    192 if drop_chr_start==True:
    193         posdic_min =  sumstats.groupby(chrom)[pos].min()

File [~/mambaforge/envs/gwas/lib/python3.8/site-packages/pandas/core/series.py:4433](https://file+.vscode-resource.vscode-cdn.net/Users/username/git/data/gwas/~/mambaforge/envs/gwas/lib/python3.8/site-packages/pandas/core/series.py:4433), in Series.apply(self, func, convert_dtype, args, **kwargs)
   4323 def apply(
   4324     self,
   4325     func: AggFuncType,
   (...)
   4328     **kwargs,
   4329 ) -> DataFrame | Series:
   4330     """
   4331     Invoke function on values of Series.
   4332 
   (...)
   4431     dtype: float64
   4432     """
-> 4433     return SeriesApply(self, func, convert_dtype, args, kwargs).apply()

File [~/mambaforge/envs/gwas/lib/python3.8/site-packages/pandas/core/apply.py:1088](https://file+.vscode-resource.vscode-cdn.net/Users/username/git/data/gwas/~/mambaforge/envs/gwas/lib/python3.8/site-packages/pandas/core/apply.py:1088), in SeriesApply.apply(self)
   1084 if isinstance(self.f, str):
   1085     # if we are a string, try to dispatch
   1086     return self.apply_str()
-> 1088 return self.apply_standard()

File [~/mambaforge/envs/gwas/lib/python3.8/site-packages/pandas/core/apply.py:1143](https://file+.vscode-resource.vscode-cdn.net/Users/username/git/data/gwas/~/mambaforge/envs/gwas/lib/python3.8/site-packages/pandas/core/apply.py:1143), in SeriesApply.apply_standard(self)
   1137         values = obj.astype(object)._values
   1138         # error: Argument 2 to "map_infer" has incompatible type
   1139         # "Union[Callable[..., Any], str, List[Union[Callable[..., Any], str]],
   1140         # Dict[Hashable, Union[Union[Callable[..., Any], str],
   1141         # List[Union[Callable[..., Any], str]]]]]"; expected
   1142         # "Callable[[Any], Any]"
-> 1143         mapped = lib.map_infer(
   1144             values,
   1145             f,  # type: ignore[arg-type]
   1146             convert=self.convert_dtype,
   1147         )
   1149 if len(mapped) and isinstance(mapped[0], ABCSeries):
   1150     # GH#43986 Need to do list(mapped) in order to get treated as nested
   1151     #  See also GH#25959 regarding EA support
   1152     return obj._constructor_expanddim(list(mapped), index=obj.index)

File [~/mambaforge/envs/gwas/lib/python3.8/site-packages/pandas/_libs/lib.pyx:2870](https://file+.vscode-resource.vscode-cdn.net/Users/username/git/data/gwas/~/mambaforge/envs/gwas/lib/python3.8/site-packages/pandas/_libs/lib.pyx:2870), in pandas._libs.lib.map_infer()

File [~/mambaforge/envs/gwas/lib/python3.8/site-packages/gwaslab/quickfix.py:190](https://file+.vscode-resource.vscode-cdn.net/Users/username/git/data/gwas/~/mambaforge/envs/gwas/lib/python3.8/site-packages/gwaslab/quickfix.py:190), in _quick_assign_i_with_rank..(x)
    187     posdiccul[i]= posdiccul[i-1] + posdiccul[i] + sumstats[pos].max()*chrpad
    189 # convert base pair postion to x axis position using the cumulative sum dictionary
--> 190 sumstats["add"]=sumstats[chrom].apply(lambda x : posdiccul[int(x)-1])
    192 if drop_chr_start==True:
    193         posdic_min =  sumstats.groupby(chrom)[pos].min()
```

**Comment by Cloufield** [comment]
I can get the same error after adding a variant with CHR=0 into the sumstats for testing. I guess the reason might be that some values in CHR are out of range. Could you please provide the last row of your error log (The line of `keyerror: -1` ) to verify it?  Thanks a lot. 


![image](https://github.com/Cloufield/gwaslab/assets/40289485/c6a9800c-407e-4d0c-ab0f-edd68ed7369b)

**Comment by swvanderlaan** [comment]
I think you are right: there are chromosomes set as '0' I noticed. These are not filtered out somehow? Which of the utilities could I use for this?

**Comment by Cloufield** [comment]
I will add a range check for chromosomes in the next version (currently gwaslab only check the notation and data type but not range). 

For the current version, after `.basic_check()`, you can manually filter in the data using `sumstats.filter_value()`  

Example 1, filter out variants with CHR <=0 and keep CHR >0 in place.
```
mysumstats.basic_check()
mysumstats.filter_value('CHR > 0', inplace=True)
```

Example 2, filter and plot using chain call.
```
mysumstats.basic_check()
mysumstats.filter_value('CHR > 0').plot_mqq(...) 
```

**Comment by Cloufield** [comment]
Hi, 
this bug is fixed in v3.4.20. Thanks for your patience.
